modules/schema/src/main/java/org/apache/ignite/internal/schema/SchemaDescriptor.java

```diff
@@ -102,6 +104,8 @@ public SchemaDescriptor(int ver, Column[] keyCols, String @Nullable[] colocation
                     colMap.put(c.name(), c);
                 });
 
+        this.physicalOrderMatchesLogical = colMap.values().stream().allMatch(col -> col.columnOrder() == col.schemaIndex());
```

Comment:
not yours, but some strings upper seems we need to call :  colMap = newLinkedHashMap(keyCols.length + valCols.length);

Reply:
fixed